File: 0225/solution.py
from collections import deque


class MyStack:
    def __init__(self):
        self.stack = deque()

    def push(self, x: int) -> None:
        self.stack.append(x)
        for _ in range(len(self.stack) - 1):
            self.stack.append(self.stack.popleft())

    def pop(self) -> int:
        return self.stack.popleft()

    def top(self) -> int:
        return self.stack[0]

    def empty(self) -> bool:
        return True if not self.stack else False

File: 0225/solution.java
import java.util.Queue;
import java.util.LinkedList;

class MyStack {
    Queue<Integer> first;
    Queue<Integer> second;

    public MyStack() {
        first = new LinkedList<>();
        second = new LinkedList<>();
    }

    public void push(int x) {
        while (!first.isEmpty()) {
            second.offer(first.poll());
        }
        first.offer(x);
        while (!second.isEmpty()) {
            first.offer(second.poll());
        }
    }

    public int pop() {
        return first.poll();
    }

    public int top() {
        return first.peek();
    }

    public boolean empty() {
        return first.isEmpty();
    }
}
